src/bus/webSocketConnection.spec.js

```diff
@@ -442,6 +325,176 @@ describe('Bus/WebSocketConnection', function() {
     });
   });
 
+  describe('onError', function() {
+    context('when the websocket throws an error', function() {
+      let expectedOrganization;
+      let onError;
+      let ws;
+      beforeEach(function() {
+        expectedOrganization = fixture.build('organization');
+
+        ws = new WebSocketConnection(
+          expectedWebSocket,
+          expectedOrganization.id
+        );
+
+        ws._jsonRpcId = faker.random.uuid();
+        ws._messageHandlers = {
+          [faker.random.uuid()]: this.sandbox.stub()
+        };
+
+        onError = this.sandbox.spy(ws, 'onError');
+      });
+
+      it('throws an error', function() {
+        try {
+          webSocketServer.emit('error');
+        } catch (err) {
+          expect(onError).to.throw(Error);
+        }
+      });
+
+      it('resets the jsonRpcId', function() {
+        try {
+          webSocketServer.emit('error');
+        } catch (err) {
+          expect(ws._jsonRpcId).to.be.null;
+        }
+      });
+
+      it('resets the messageHandlers', function() {
+        try {
+          webSocketServer.emit('error');
+        } catch (err) {
+          expect(ws._messageHandlers).to.be.empty;
+        }
+      });
+    });
+  });
+
+  describe('onMessage', function() {
+    context('when message handlers exist for the message id', function() {
+      context('when the message is a valid json string', function() {
+        let expectedMessageHandlers;
+        let expectedOrganization;
+        let expectedUUID;
+        let ws;
+
+        beforeEach(function() {
+          expectedOrganization = fixture.build('organization');
+          expectedUUID = faker.random.uuid();
+          expectedMessageHandlers = {
+            [expectedUUID]: this.sandbox.stub()
+          };
+
+          ws = new WebSocketConnection(
+            expectedWebSocket,
+            expectedOrganization.id
+          );
+
+          ws._messageHandlers = expectedMessageHandlers;
+
+          webSocketServer.emit(
```

Comment:
What do you think about just invoking `onMessage`? The other, previous cases (that also have comments about this same thing) were more of a call + response test because we had to have an open connection for it to work